src/sage/matrix/matrix2.pyx

```diff
@@ -18656,8 +18669,14 @@ def _matrix_power_symbolic(A, n):
         # D^i(f) / i! with f = x^n and D = differentiation wrt x
         if hasattr(mk, 'radical_expression'):
             mk = mk.radical_expression()
-        vk = [(binomial(n, i) * mk**(n-i)).simplify_full()
-              for i in range(nk)]
+
+        # If mk=0 return mk^(n-i) instead of mk**(n-i)
+        if mk:
+            vk = [(binomial(n, i) * mk._pow_(n-i)).simplify_full()
+                  for i in range(nk)]
+        else:
+            vk = [(binomial(n, i).simplify_full() * mk._pow_(n-i))
```

Comment:
In fact, a more suitable result would be given by             
`vk = [(binomial(n, i).simplify_full() * kronecker_delta(n,i))
                  for i in range(nk)]`
This even allows to evaluate with small values of n